common/persistence/visibility/visiblity_manager_metrics.go

```diff
@@ -180,7 +181,8 @@ func (m *visibilityManagerMetrics) updateErrorMetric(scope metrics.Scope, err er
 		scope.IncCounter(metrics.VisibilityPersistenceTimeout)
 		scope.IncCounter(metrics.VisibilityPersistenceFailures)
 	case *serviceerror.ResourceExhausted:
-		scope.IncCounter(metrics.VisibilityPersistenceResourceExhausted)
+		scope.Tagged(metrics.ResourceExhaustedCauseTag(enumspb.RESOURCE_EXHAUSTED_CAUSE_PERSISTENCE_RATE_LIMIT)).
```

Comment:
Did you double check that `metrics.VisibilityPersistenceResourceExhausted` counter is not reported anywhere else w/o `metrics.ResourceExhaustedCauseTag` tag? And for other metrics too?

Reply:
Yes, I found all references to the resource_exhausted metrics.